lib/parachain/validation_protocol.go

```diff
@@ -8,23 +8,199 @@ import (
+}
+
+// New returns new ApprovalDistributionMessage VDT
+func (ApprovalDistribution) New() ApprovalDistribution {
+	return NewApprovalDistributionVDT()
+}
+
+func (ApprovalDistribution) Index() uint {
+	return 4
+}
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (ad *ApprovalDistribution) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*ad)
+	err = vdt.Set(val)
+	if err != nil {
+		return fmt.Errorf("setting value to varying data type: %w", err)
+	}
+
+	*ad = ApprovalDistribution(vdt)
+	return nil
+}
+
+type StatementDistribution StatementDistributionMessage
+
+func NewStatementDistributionVDT() StatementDistribution {
+	vdt, err := scale.NewVaryingDataType(SignedFullStatement{}, SecondedStatementWithLargePayload{})
+	if err != nil {
+		panic(err)
+	}
+	return StatementDistribution(vdt)
+}
+
+func (StatementDistribution) New() StatementDistribution {
+	return NewStatementDistributionVDT()
+}
+
+// Value returns the value from the underlying VaryingDataType
+func (sd *StatementDistribution) Value() (scale.VaryingDataTypeValue, error) {
+	vdt := scale.VaryingDataType(*sd)
+	return vdt.Value()
+}
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (sd *StatementDistribution) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*sd)
+	err = vdt.Set(val)
+	if err != nil {
+		return fmt.Errorf("setting value to varying data type: %w", err)
+	}
+
+	*sd = StatementDistribution(vdt)
+	return nil
+}
+
+func (StatementDistribution) Index() uint {
+	return 3
+}
+
+type ValidationProtocol scale.VaryingDataType
+
+func NewValidationProtocolVDT() ValidationProtocol {
+	vdt, err := scale.NewVaryingDataType(BitfieldDistribution{}, StatementDistribution{}, ApprovalDistribution{})
+	if err != nil {
+		panic(err)
+	}
+	return ValidationProtocol(vdt)
+}
+
+// New returns new ApprovalDistributionMessage VDT
+func (ValidationProtocol) New() ValidationProtocol {
+	return NewValidationProtocolVDT()
+}
+
+// Value returns the value from the underlying VaryingDataType
+func (vp *ValidationProtocol) Value() (scale.VaryingDataTypeValue, error) {
+	vdt := scale.VaryingDataType(*vp)
+	return vdt.Value()
+}
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (vp *ValidationProtocol) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*vp)
+	err = vdt.Set(val)
+	if err != nil {
+		return fmt.Errorf("setting value to varying data type: %w", err)
+	}
+
+	*vp = ValidationProtocol(vdt)
+	return nil
+
 }
 
 // Type returns ValidationMsgType
-func (*ValidationProtocolV1) Type() network.MessageType {
+func (ValidationProtocol) Type() network.MessageType {
 	return network.ValidationMsgType
 }
 
 // Hash returns the hash of the CollationProtocolV1
-func (vp *ValidationProtocolV1) Hash() (common.Hash, error) {
+func (vp *ValidationProtocol) Hash() (common.Hash, error) {
```

Comment:
I think this would work even without making ValidationProtocol pointer

Reply:
I think our linter gets mad if we have a mix of pointer and non pointer receivers

Reply:
> I think our linter gets mad if we have a mix of pointer and non pointer receivers

Yeah, that's true. I think I might have meant to change from pointer to non-pointer in all it's methods. But if linter gets mad, no need to address this comment. Anyway this comment ranges from super nitpicky to personal preference. :laughing: